Source/d_loop.c

```diff
@@ -219,7 +219,14 @@ void NetUpdate (void)
     NET_SV_Run();
 
     // check time
+    if (fastdemo)
+    {
+      nowtime = I_GetTime() / ticdup;
```

Comment:
Doesn't this collide (or is redundant) with the code in `i_system.c:I_Init()`? Does the `-speed` parameter currently work?

Reply:
You're right, the `-speed` parameter doesn't work.

Reply:
`I_GetTime()` is a function pointer. I believe it is always correct to use it. What it actually means is set in `I_Init()`.

Reply:
So we move `GetAdjustedTime()` to `I_Init()`? `GetAdjustedTime()` is required for newsync mode. Alternatively, we can use another global variable for `-speed` parameter (there is `realtic_clock_rate`).

Reply:
Does the demo warp feature work now?

Reply:
Yes, now I remember: I couldn't get `demowarp` and `demoskip` to work.